### PR TITLE
add TcpStream::peek

### PR DESCRIPTION
**Pull request by seanmonstar**

### CODE REVIEW

File: src/sys/windows/tcp.rs [hunk 1]
```diff
@@ -252,6 +252,44 @@ impl TcpStream {
         }
     }
 
+    pub fn peek(&self, buf: &mut [u8]) -> io::Result<usize> {
+        let mut me = self.inner();
+
+        match me.read {
```
Comment (carllerche) [inline src/sys/windows/tcp.rs]: It looks like this block is copied verbatim from `readv`? Would you mind extracting this to a `before_read` fn or something like that?

Comment (seanmonstar) [reply src/sys/windows/tcp.rs]: Done.

**Comment by seanmonstar** [comment]
Ah, so this requires a newer version of Rust than what is being tested on Travis. Breaking change needing 0.7, or just Deal With It?

**Comment by seanmonstar** [comment]
CI passed!

**Pull request merged**